docs/sources/reference/builder.md

```diff
@@ -350,8 +350,9 @@ accessible from the host by default. To expose ports to the host, at runtime,
 
 The `ENV` instruction sets the environment variable `<key>` to the value
 `<value>`. This value will be passed to all future 
-`RUN`, `ENTRYPOINT`, and `CMD` instructions. This is
-functionally equivalent to prefixing the command with `<key>=<value>`
+`RUN`, `ENTRYPOINT`, and `CMD` instructions. To set a value for a single
+command without affecting future commands or containers, prefix the command
+with `<key>=<value>`.
```

Comment:
I know this isn't really due to your proposed change, but since you're here anyway....  :-)   perhaps you could tweak this paragraph a little more.  Its a bit misleading to say that it just impacts RUN, ENTRYPOINT and CMD because I believe that @erikh made it so that these env vars are also used in variable substitutions as well. See https://github.com/docker/docker/blob/master/builder/evaluator.go#L46 for the list of commands that we do env var substitutions.  If you don't want to make this change let me know and I'll do it in a different PR.

As for your text, you may need to make it clear that prefixing only works for the sh version, not the JSON version (I think).